src/main.rs

```diff
@@ -377,21 +377,26 @@ fn main() -> ExitCode {
             }
         };
 
-        // TODO: borrow only here
-        let compile_id_dir = e
+        let compile_id_dir: PathBuf = e
             .compile_id
-            .clone()
-            .map_or(format!("unknown_{}", lineno), |e: CompileId| {
-                format!("{}_{}_{}", e.frame_id, e.frame_compile_id, e.attempt)
-            });
+            .as_ref()
+            .map_or(
+                format!("unknown_{lineno}"),
+                |CompileId {
+                     frame_id,
+                     frame_compile_id,
+                     attempt,
+                 }| { format!("{frame_id}_{frame_compile_id}_{attempt}") },
```

Comment:
huh, f-string style is nice

Reply:
Just be aware that it only works with identifiers - so you can't do:
```
format!("{foo.bar}")
```
but you can do:
```
format!("{myvar}", myvar = foo.bar)
```